contrib/zpages/src/main/java/io/opencensus/contrib/zpages/StatszZPageHandler.java

```diff
@@ -193,61 +204,64 @@ private void emitDirectoryTable(
       /*@Nullable*/ String path,
       PrintWriter out,
       Formatter formatter) {
-    out.write("<h2>Views</h2>");
-    if (currentNode == null) {
-      formatter.format(
-          "<p><font size=+2>Directory not found: %s. Return to root.</font></p>", path);
-      currentNode = root;
-    }
-    if (currentNode == root || path == null) {
-      path = "";
-    }
-    emitDirectoryHeader(path, out, formatter);
-    out.write("<table frame=box cellspacing=0 cellpadding=2>");
-    for (Entry<String, TreeNode> entry : currentNode.children.entrySet()) {
-      TreeNode child = entry.getValue();
-      String relativePath = entry.getKey();
-      if (child.view == null) { // Directory node, emit a row for directory.
-        formatter.format(
-            "<tr class=\"%s\"><td>Directory: <a href='?%s=%s'>%s</a> (%d %s)</td></tr>",
-            CLASS_LARGER_TR,
-            QUERY_PATH,
-            path + '/' + relativePath,
-            relativePath,
-            child.views,
-            child.views > 1 ? "views" : "view");
-      } else { // View node, emit a row for view.
-        String viewName = child.view.getName().asString();
+    synchronized (monitor) {
```

Comment:
Why this? You call this function while you hold the monitor lock correct? You need to add an annotation on the method that is guarded by the monitor.

Reply:
Oh that's right - updated.